### PR TITLE
Higher Power: updated homepage, using homepage.json

### PR DESCRIPTION
**Pull request by breville**
This implements a new Higher Power-themed https://code.org/ homepage hero, using a new `homepage.json` file.  

If the file is removed, the homepage falls back to the existing hero.  The `environments` and/or `dcdo` properties can also be used to declare whether the hero is seen.

Going forward, content editors can edit `homepage.json` to author the homepage hero, with no developer intervention required.  That said, the file has limited capabilities which can be expanded going forward.

The support for `homepage.json` is based on [the support ](https://github.com/code-dot-org/code-dot-org/blob/edd7cb3e9c477d1bd752a3baa1087c4409abdc0c/dashboard/lib/announcements.rb)for `announcements.json` in dashboard.

## Additional notes on `homepage.json`

Features include:
- If the file doesn't exist, it falls back to the existing homepage implemented in code.
- Based on `announcements.json` it includes familiar features such as the ability to declare multiple heroes while only one is active.
- You can specify one background image.
- You can specify a series of "items" like text or images.  The Higher Power one demonstrates one of each.
- You can specify a series of "actions" which map to our usual call-to-action buttons.
- Text comes from the `i18n` gsheet which means it's loc'ed.
- You can specify a list of environments (like `"staging"`, `"test"`, `"production"`) if you want to try out a hero in a subset of them first.
- You can specify a DCDO flag that's required for a hero to show.

### desktop
Screen Shot 2022-02-09 at 8 24 58 AM

### mobile
Screen Shot 2022-02-09 at 8 25 15 AM

### CODE REVIEW

**Review by epeach — COMMENTED** [review]
Pegasus is a bit of a black box for me. So with my small knowledge, this looks good, but I'd love if someone with more expertise in this area took a look too!

**Review by KylieModen — COMMENTED** [review]
When this PR is merged, will the homepage banner then change? If so we should wait for 2/16 DTP

**Comment by breville** [comment]
> When this PR is merged, will the homepage banner then change? If so we should wait for 2/16 DTP

Good question.  That was going to be the plan, but this inspired me to add an optional value for each hero that looks like this:

```
"environments": ["development", "staging", "test"],
```

If this value is present, the system will only show the hero if the current environment is included in the list.  If this value is not present, then the default hero will be shown.

This way, we can get this hero showing on non-production environments ahead of the official launch.

**Review by molly-moen — APPROVED** [review]
Nice!

**Review by maureensturgeon — COMMENTED** [review]
Would you be able to include a screenshot on the PR description of what the resulting change looks like on desktop and mobile? Thanks!

File: pegasus/sites.v3/code.org/homepage.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      "environments": ["development", "staging", "test"],
+      "desktopImage": "/images/marketing/homepage-higher-power-background.png",
+      "items": [
+        {
+          "type": "image",
+          "url": "/images/marketing/homepage-higher-power-logo.png"
+        },
+        {
+          "type": "text",
+          "text": "homepage_higher_power_body"
```
Comment (maureensturgeon) [inline pegasus/sites.v3/code.org/homepage.json:17]: Does marketing have the ability/ know the process around adding text to the i18n files?

Comment (breville) [reply pegasus/sites.v3/code.org/homepage.json:17]: That's a great question, and I hope the answer is yes.  It's one of our oldest gsheets for content editors, and certainly in the past we expected content editors to be able to work with it.  I'll be available to help anybody who needs it.

**Review by maureensturgeon — APPROVED** [review]
Looks great!

File: pegasus/src/homepage.rb [hunk 2]
```diff
@@ -1,6 +1,71 @@
...
+    # If the banner has a required DCDO flag, then it must be set.
+    return nil if banner["dcdo"] && !DCDO.get(banner["dcdo"], false)
+
+    banner ? banner.merge({"id": banner_id_for_page}) : nil
+  end
+
+  def self.load_announcements
+    # Reloads JSON file with announcement data on each page load
+    # in non-production environments
+    unless (@@load_error || @@loaded) && rack_env?(:production)
```
Comment (breville) [inline pegasus/src/homepage.rb:32]: @molly-moen I think I need to remove the `&& rack_env?(:production)` if I want to be able to change the announcement via DCDO on production?

Comment (molly-moen) [reply pegasus/src/homepage.rb:32]: Yeah I think so. If it's not there then I believe it will only be changeable on a dtp.

**Pull request merged**